server/service/src/invoice/outbound_return/delete/mod.rs

```diff
@@ -0,0 +1,211 @@
+use repository::{ActivityLogType, InvoiceRowRepository, RepositoryError, TransactionError};
+
+pub mod validate;
+
+use validate::validate;
+
+use crate::{
+    activity_log::activity_log_entry,
+    invoice::common::get_lines_for_invoice,
+    invoice_line::stock_out_line::{
+        delete::{delete_stock_out_line, DeleteStockOutLine, DeleteStockOutLineError},
+        StockOutType,
+    },
+    service_provider::ServiceContext,
+    WithDBError,
+};
+
+type OutError = DeleteOutboundReturnError;
+
+pub fn delete_outbound_returns(
+    ctx: &ServiceContext,
+    ids: Vec<String>,
+) -> Result<Vec<String>, OutError> {
+    ctx.connection
+        .transaction_sync(|connection| -> Result<(), OutError> {
```

Comment:
Because of the for loop, I had to annotate the return type here... am I missing something for Rust to infer the type or is this expected?

Reply:
I don't think it's because of the for loop (at least getting rid of the for loop didn't fix the error for me)
I think it's because you have several different error types returned from `?` calls in the `transaction_sync`.

If all the `?` returns are the same type it can infer the type, but it doesn't seem to infer a type that they can all reach via a `From` implementation. I suspect it might work if the `transaction_sync` function didn't wrap the returned error in it's on generic.

This also works to fix the problem (as an alertnative approach)
`        .map_err(|error: TransactionError<DeleteOutboundReturnError>| error.to_inner_error())?;`

Reply:
this is probably the cleverest typo that I've seen!
`alertnative` actually makes sense.. this is the 'alert native' approach 😆